src/msg/msg_types.h

```diff
@@ -175,21 +179,34 @@ WRITE_CLASS_ENCODER(ceph_sockaddr_storage)
 /*
  * encode sockaddr.ss_family as network byte order
  */
+static inline size_t get_sockaddr_length(int family) {
+    switch (family) {
+    case AF_INET:
+      return sizeof(struct sockaddr_in);
+    case AF_INET6:
+      return sizeof(struct sockaddr_in6);
+    }
+    return sizeof(struct sockaddr_storage);
+}
+
 static inline void encode(const sockaddr_storage& a, bufferlist& bl) {
 #if defined(__linux__)
   struct sockaddr_storage ss = a;
   ss.ss_family = htons(ss.ss_family);
   ::encode_raw(ss, bl);
 #elif defined(__FreeBSD__) || defined(__APPLE__)
+  // We do not copy ss_len into the buffer since the on-wire format 
+  // is compatible with the Linux sockaddr
   ceph_sockaddr_storage ss{};
   auto src = (unsigned char const *)&a;
   auto dst = (unsigned char *)&ss;
   src += sizeof(a.ss_len);
   ss.ss_family = a.ss_family;
   src += sizeof(a.ss_family);
   dst += sizeof(ss.ss_family);
-  const auto copy_size = std::min((unsigned char*)(&a + 1) - src,
-				  (unsigned char*)(&ss + 1) - dst);
+  // const auto copy_size = std::min((unsigned char*)(&a + 1) - src,
```

Comment:
why the existing `copy_size` is wrong?

Reply:
@tchaikov 
Because it depends on how the variables are layout out on the stack?
Besides the fact that comparing pointers is sort of trickery, which I usually don't really like since it is hard to follow. What I'm suggesting is a more logic evaluation of what is going on.

Reply:
> Because it depends on how the variables are layout out on the stack? 

this does not render it *wrong*.